### PR TITLE
test-event: do not work in assert()

### PR DESCRIPTION
**Pull request by yuwata**
Follow-up for 85e55d14dea66f5fe412ca8128487d5ea828b7b1.

C.f. comments in #7271.

Also, tests for empty string are added to test-parse-util.

### CODE REVIEW

File: src/libsystemd/sd-event/test-event.c [hunk 1]
```diff
@@ -317,11 +317,11 @@ static void test_rtqueue(void) {
...
-        assert(sigqueue(getpid_cached(), SIGRTMIN+2, (union sigval) { .sival_int = 1 }) >= 0);
-        assert(sigqueue(getpid_cached(), SIGRTMIN+3, (union sigval) { .sival_int = 2 }) >= 0);
-        assert(sigqueue(getpid_cached(), SIGUSR2, (union sigval) { .sival_int = 3 }) >= 0);
-        assert(sigqueue(getpid_cached(), SIGRTMIN+3, (union sigval) { .sival_int = 4 }) >= 0);
-        assert(sigqueue(getpid_cached(), SIGUSR2, (union sigval) { .sival_int = 5 }) >= 0);
+        assert_se(sigqueue(getpid_cached(), SIGRTMIN+2, (union sigval) { .sival_int = 1 }) >= 0);
+        assert_se(sigqueue(getpid_cached(), SIGRTMIN+3, (union sigval) { .sival_int = 2 }) >= 0);
+        assert_se(sigqueue(getpid_cached(), SIGUSR2, (union sigval) { .sival_int = 3 }) >= 0);
+        assert_se(sigqueue(getpid_cached(), SIGRTMIN+3, (union sigval) { .sival_int = 4 }) >= 0);
+        assert_se(sigqueue(getpid_cached(), SIGUSR2, (union sigval) { .sival_int = 5 }) >= 0);
```
Comment (evverx) [inline src/libsystemd/sd-event/test-event.c:324]: I think the following commands could help to fix the rest of the tests:
```sh
cat <<'EOF' >coccinelle/assert.cocci
@@ expression E; @@

-assert(E)
+assert_se(E)
EOF

meson test --list -C ./build/ |
xargs -I'%%TEST%%' -n1 find src -name '%%TEST%%*' |
xargs spatch --sp-file coccinelle/assert.cocci --in-place
```
especially `test-ndisc-rc` which is currently failing when being compiled with `-Db_ndebug=true`

**Comment by yuwata** [comment]
@evverx Updated. test-ndisc-rs is fixed. But I've not added coccinelle rules yet.

**Comment by poettering** [comment]
excellent! thanks!
